packages/web-components/src/components/va-memorable-date/va-memorable-date.tsx

```diff
@@ -94,19 +99,36 @@ export class VaMemorableDate {
     const daysForSelectedMonth = monthNum > 0 ? days[monthNum] : [];
     const leapYear = checkLeapYear(yearNum);
     // Check validity of date if invalid provide message and error state styling
-    if (
+    // Set the state so that we can set aria-invalid on the right component
+    this.invalidYear = (
+      !year ||
       yearNum < minYear ||
-      yearNum > maxYear ||
+      yearNum > maxYear
+    );
+
+    this.invalidMonth = (
+      !month ||
       monthNum < minMonths ||
-      monthNum > maxMonths ||
+      monthNum > maxMonths
+    );
+
+    this.invalidDay =  (
+      !day ||
       dayNum < minMonths ||
-      dayNum > daysForSelectedMonth.length ||
-      day === '' ||
-      month === '' ||
-      year === '' ||
-      (!leapYear && monthNum === 2 && dayNum > 28) ||
-      (this.required && !isFullDate(this.value))
```

Comment:
I'm not quite sure why we were checking `this.required && !isFullDate(this.value)` in the first place since missing `day`, `month`, _or_` year` would result in the same thing happening regardless of whether or not it was required. While reviewing, try to think if not checking `required` would cause any regressions.

Reply:
Are you saying that if a memorable date component was not required and someone left it blank, they would get the same error messages?

Reply:
I was curious, so I added a couple of tests around the required prop just to see what happens.

On va-date, I found that even when a date is not required, we show an error message if someone clicks into the component and then tabs out without entering anything. This test fails.
```
it('does not display an error message when required is false and input is left blank', async () => {
      const page = await newE2EPage();
      await page.setContent(
        '<va-date name="test" />',
      );
      const date = await page.find('va-date');
      const handleYear = await page.$('pierce/[name="testYear"]');
      await handleYear.click({ clickCount: 1 });
      // Trigger Blur
      await handleYear.press('Tab');
  
      await page.waitForChanges();
      expect(date.getAttribute('error')).toEqual('');
    })
```

I also added this test to the memorable date component and it failed, meaning we're not doing any checking when a date is required on this component, only when it is incorrect.
```
  it('displays an error message when required is true and input is left blank', async() => {
    const page = await newE2EPage();
    await page.setContent(
      '<va-memorable-date name="test" required="true" />',
    );
    const elementYear = await page.find('va-memorable-date >>> .input-year');
    const handleYear = await page.$('pierce/[name="testYear"]');
    await handleYear.click({ clickCount: 1 });
    // Trigger Blur
    await handleYear.press('Tab');

    await page.waitForChanges();
    expect(elementYear.getAttribute('error')).toEqual('Please enter a valid date');
  });
```

It would be a good idea to add a couple of tests around the required prop for both of these components, one testing what happens when required is false and the input is left blank and one testing what happens when required as true and the input is left blank.

Reply:
This commit changes the behavior a bit and adds some tests: https://github.com/department-of-veterans-affairs/component-library/pull/489/commits/6718483cf48118b6cb1a6ece5bd98746e8342c66